snewpdag/values/TimeSeries.py

```diff
@@ -1,158 +1,81 @@
+  def add(self, times):
     """
-    times:  an array of times.  Subtract reference time before appending.
-      s        a single time (float)
-      [s1,s2]  two times (floats)
-      (s,ns)   a single (s,ns) - specifically needs to be a tuple!
-      [(s1,ns1),(s2,ns2)]  two times (s,ns)
+    Add times to series.
+    times:  an array of timestamps, assumed to be seconds unless
+            it's an array of Quantity, in which case convert to seconds.
     """
-    ts = np.array(times)
-    shape = np.shape(ts)
-    if (len(shape) == 2 and shape[1] == 2) or \
-        (shape == (2,) and isinstance(times, tuple)):
-      # array of (s,ns)
-      tt = ts
-    elif shape == () or len(shape) == 1:
-      # array or scalar of s in float
-      tt = time_tuple_from_float(ts)
+    ts = times.to(u.s) if hasattr(times, 'unit') else times
+    if self.start == self.stop: # also includes both being None
+      self.times = np.append(self.times, ts)
     else:
-      logging.error("input array has wrong shape {}".format(shape))
-      return
-    d = subtract_time(tt, self.reference)
-    t = np.multiply(d[...,0], ns_per_second, dtype=np.int64)
-    t = np.add(t, d[...,1], dtype=np.int64)
-    self.add_offsets(t)
+      m = np.full_like(ts, True) if self.start == None else (ts >= self.start)
+      if self.stop != None:
+        m &= (ts < self.stop)
+      self.times = np.append(self.times, ts[m])
 
-  def event(self, index):
-    """
-    get the normalized (s,ns) time of indexed event(s).
-    if index is a simple number, just return one result.
-    if index is an array of indices, return corresponding results in array.
-    """
-    try:
-      if np.isscalar(index):
-        t1 = np.add(self.reference, (0, self.times[index]))
-      else:
-        i = np.array(index)
-        t0 = np.full((len(i), 2), self.reference)
-        dt = np.column_stack((np.zeros(len(i)), self.times[i]))
-        t1 = t0 + dt
-    except IndexError:
-      logging.error('TimeSeries: index out of bounds')
-      return None
-    else:
-      return normalize_time(t1)
-
-  def histogram(self, nbins, duration=0, start=()):
+  def histogram(self, nbins, start=None, stop=None):
     """
     Make a histogram out of the time series.
-    The histogram's low edge will be the start time.
-    The defaults for duration and start are those of the TimeSeries itself.
-    if ref time is 100s, but requested start time is 90s, then
-    all offsets need to add 10s before making the histogram.
-
-    Returns histogram, bin edges
     """
-    width = self.duration_ns if duration == 0 else duration * ns_per_second
-    t0 = self.start if start == () else time_tuple_from_field(start)
-    offset = offset_from_time_tuple(subtract_time(self.reference, t0))
-    h, edges = np.histogram(self.times + offset, bins=nbins, range=(0, width))
-    return h
+    t0 = self.start if start == None else start
+    t1 = self.stop if stop == None else stop
+    if t0 == None:
```

Comment:
Since TimeSeries just contains a bunch of timestamps, in principle one doesn't have to have limits.  On the other hand, if it makes life simpler to always have limits, open to change.